test_pos/app.html

```diff
@@ -2,67 +2,209 @@
+	  }
+
+          function inputChange(event) {
+              updateQr();
+	  }
+
+          window.onload = function() {
+	    Particles.init({
+	      selector: '.particle-section',
+              maxParticles: 180,
+              connectParticles: true,
+	    });
+            document.querySelector('#input-amount').addEventListener('input', inputChange);
+            document.querySelector('#input-invoiceid').addEventListener('input', inputChange);
+	  };
+
+	  function nonce() {
+	      return Math.floor(new Date().getTime() / 1000);
+	  }
+
+	  function sign(data) {
+	      var hash = CryptoJS.HmacSHA256(data, apisecret);
+	      return CryptoJS.enc.Base64.stringify(hash);
+	  }
+
+	  async function postPayDb(endpoint, params) {
+	      params['api_key'] = apikey;
+	      params['nonce'] = nonce();
+	      var body = JSON.stringify(params);
+	      console.log(body);
+
+	      var sig = sign(body);
+	      console.log(sig);
+
+	      const response = await fetch(URL_BASE + endpoint, {
+		  method: 'POST',
+		  headers: {'Content-Type': 'application/json', 'X-Signature': sig},
+		  body: body
+	      });
+	      console.log(response.status);
+	  
+	      return await response.json();
+	  }
+
+	  socket.on('connect', () => {
+	      console.log('socket connected, id', socket.id);
+	      // create auth data
+	      var nonce_ = nonce();
+	      var sig = sign(nonce_.toString());
+	      auth = {signature: sig, api_key: apikey, nonce: nonce_};
+	      // emit auth message
+	      socket.emit('auth', auth);
+	  });
+
+	  socket.on("info", (arg) => {
+	      console.log(arg);
+	  });
+	  
+	  socket.on('tx', (arg) => {
+	      console.log(arg);
+              var tx = JSON.parse(arg);
+              if (tx.recipient == email) {
+                console.log("recieved");
+		Swal.fire('Transaction Received!', arg, 'success').then(
+                  function(result) {
+                    if(result.isConfirmed) {
+                      Swal.fire(
+                        {
+                          title: "give rebate",
+                          html: `<input type='email' id='email-input' class='swal2-input' placeholder='email'>`,
+                          preConfirm: function() {
+                            const rebateEmail = Swal.getPopup().querySelector("#email-input").value;
+                            if(!rebateEmail) {
+                              Swal.showValidationMessage('please enter an email');
+                            }
+                            return rebateEmail; 
+                          }
+                        }
+                      ).then(
+                          function(emailInput) {
+                            postPayDb('payment_create', {reason: "rebate", recipient: emailInput.value, amount: Math.floor(tx.amount * REBATE_FACTOR), category: "testing", message: 1})
+                            .then(
+                              function(res) {
+                                console.log("sent");
```

Comment:
you should validate the amount and invoice id, also show a nice popup to show the rebate succeeded